src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/TestUtil.java

```diff
@@ -237,4 +255,23 @@ public static int adjustFlushCountForOutputStreamClose(int flushCount) {
         return result;
     }
 
+    /**
+     * Verifies that a task with an expected result can be executed with
+     * limited stack size and time duration.
+     *
+     * @param <V> return type of task
+     * @param callable the task to execute
+     * @return true if task completes
+     * @throws InterruptedException if thread execution is interrupted
+     * @throws ExecutionException if task fails
+     */
+    public static <V> boolean isExecutableWithLimitedResources(Callable<V> callable)
+            throws InterruptedException, ExecutionException {
+        final FutureTask<V> futureTask = new FutureTask<>(callable);
+        final Thread thread = new Thread(null, futureTask,
+                "LimitedStackSizeThread", MINIMAL_STACK_SIZE);
+        thread.start();
+        thread.join(EXECUTION_TIMEOUT);
+        return futureTask.get() != null;
```

Comment:
Doesn't this code seem similar to https://github.com/checkstyle/checkstyle/issues/9253#issuecomment-774761543 where I said adding it would cause issues?

Reply:
It should not cause issues, since we are checking for a result. I avoided checking for exceptions for this reason. Any exceptional behavior during execution of this thread will cause `get()` to fail.

Reply:
Also, in https://github.com/checkstyle/checkstyle/issues/9253#issuecomment-774761543, test failure was probably from not starting threads.